test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -162,24 +163,31 @@ database_operation::insert_operation(thread_context *tc)
         auto &cc = ccv[counter];
         while (tc->transaction.active() && tc->running()) {
             /* Insert a key value pair. */
-            if (!tc->insert(cc.cursor, cc.coll.id, start_key + added_count)) {
-                committed = false;
-                break;
+            bool rollback_required = tc->insert(cc.cursor, cc.coll.id, start_key + added_count);
+            if (!rollback_required)
+                added_count++;
+
+            if (tc->transaction.can_commit_rollback()) {
```

Comment:
I dont think you want to check this if `rollback_required` has been set to `true` line 166, do you ?